Python/optimizer_symbols.c

```diff
@@ -147,6 +147,36 @@ _Py_uop_sym_get_const(JitOptContext *ctx, JitOptSymbol *sym)
     return NULL;
 }
 
+_PyStackRef
+_Py_uop_sym_get_const_as_stackref(JitOptContext *ctx, JitOptSymbol *sym)
+{
+    PyObject *const_val = _Py_uop_sym_get_const(ctx, sym);
+    if (const_val == NULL) {
+        return PyStackRef_NULL;
+    }
+    return PyStackRef_FromPyObjectBorrow(const_val);
+}
+
+/*
+ Indicates whether the constant is safe to constant evaluate
+ (without side effects).
+ */
+bool
+_Py_uop_sym_is_safe_const(JitOptContext *ctx, JitOptSymbol *sym)
+{
+    PyObject *const_val = _Py_uop_sym_get_const(ctx, sym);
+    if (const_val == NULL) {
+        return false;
+    }
+    PyTypeObject *typ = Py_TYPE(const_val);
+    return (typ == &PyLong_Type) ||
+        (typ == &PyUnicode_Type) ||
+        (typ == &PyFloat_Type) ||
+        (typ == &PyDict_Type) ||
+        (typ == &PyTuple_Type) ||
+        (typ == &PyList_Type);
```

Comment:
We shouldn't constant-evaluate anything involving mutable containers. (Even `tuple` scares me a tiny bit, since it can contain arbitrary objects, but I'm pretty sure it's okay.)
```suggestion
    return (typ == &_PyNone_Type) ||
           (typ == &PyBool_Type) ||
           (typ == &PyLong_Type) ||
           (typ == &PyFloat_Type) ||
           (typ == &PyUnicode_Type) ||
           (typ == &PyTuple_Type);
```

Reply:
Looks like you missed `bytes`, I created a PR here: https://github.com/python/cpython/pull/136033